### ISSUE TITLE
Fine-tuning MODNet on a custom dataset using SOC

### ISSUE THREAD

**Issue by TatianaSnauwaert** [ZHKKKe/MODNet#206]
Hi @ZHKKKe !

Thank you for sharing this model with us.
I am trying to fine-tune the trained model using the SOC adaptation iteration on a dataset of people shot on the green screen background (180k images). I ran it for 15 epochs (batch size = 1, lr = 0.00001), the loss went down from 0.0047 to 0.0017 but visually I don't see any improvement so far.

![Screenshot from 2023-01-25 11-29-34](https://user-images.githubusercontent.com/63235607/214696150-decf96c7-67a8-4171-ad39-ab4c19d3001f.png)

Here on the left - your trained model, on the right - fine-tuned on our dataset. 

I'm trying to get rid of this green edge. Is there any parameter I should adjust to achieve it? Do you think training the model from scratch on our dataset will help improve the performance?

Any thoughts would be helpful! Let me know of you need any further details from me. 
Thank you in advance for your time! 
Best regards,
Tatiana

**Comment by ZHKKKe** [comment]
SOC is designed to fix relatively large matting errors (e.g. missing hands/ears), and it is difficult to improve boundary details. 

Calculating Foreground Color can erase the green colors on the boundaries. To calculate Foreground Color, you may use Foreground Estimation Functions from https://github.com/pymatting/pymatting .
